zcncore/wallet_base.go

```diff
@@ -980,13 +980,30 @@ func GetStakePoolUserInfo(clientID string, cb GetInfoCallback) (err error) {
 }
 
 // GetBlobbers obtains list of all active blobbers.
-func GetBlobbers(cb GetInfoCallback) (err error) {
-	if err = CheckConfig(); err != nil {
+func GetBlobbers(cb GetInfoCallback, limit, offset int, options ...bool) {
```

Comment:
We could do the API better by using functional option for all the limit, offet, and active options, so when by default we will get all active blobbers, with `GetBlobbers(cb GetInfoCallback)`, and if we only need subset with pagination, we then call it like `GetBlobbers(cb GetInfoCallback, WithOffset(100), WithLimit(20))` and if needed to get all by `WithAll()`. 

Also, we should use the `cb GetInfoCallback)` internal, the API callers should not be bothered to setup the callback function, they should only call this API be like `GetBlobber(ctx, options...) (results, error)`. 

Anyway, that's the direction we need to refactor later, I'll approve all related PRs as these changes can be done in new PRs.

Reply:
https://github.com/0chain/gosdk/issues/660

thanks for the feedback @peterlimg . Created an issue to track this.